src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractCollectorBatchBuilder.java

```diff
@@ -253,6 +259,70 @@ protected void updateCollectorBatchWithOriginEntries() {
         collectorBatch.setTotalRecords(totalRecords);
         collectorBatch.setTotalAmount(totalDebitAmount);
     }
+    
+  private void consolidateOriginEntries() {
+    List<OriginEntryFull> consolidatedOriginEntries = new ArrayList<OriginEntryFull>();
+    LOG.info("consolidateOriginEntries, starting number of origin entries: " + collectorBatch.getOriginEntries().size());
+    boolean transactionsWereConsolidated = false;
+    for (OriginEntryFull entry : collectorBatch.getOriginEntries()) {
+      OriginEntryFull matchedEntry = findMatchingOriginEntryFull(consolidatedOriginEntries, entry);
+      if (matchedEntry == null) {
+        consolidatedOriginEntries.add(entry);
+      } else {
+        LOG.debug("consolidateOriginEntries, found a transaction to consolidate: " + matchedEntry);
+        transactionsWereConsolidated = true;
+        KualiDecimal newTotal = matchedEntry.getTransactionLedgerEntryAmount()
+            .add(entry.getTransactionLedgerEntryAmount());
+        matchedEntry.setTransactionLedgerEntryAmount(newTotal);
+      }
+    }
+
+    if (transactionsWereConsolidated) {
+      Map<String, MutableInt> nextSequenceNumbers = new HashMap<>();
+      for (OriginEntryFull entry : consolidatedOriginEntries) {
+        MutableInt nextSequenceNumber = nextSequenceNumbers.computeIfAbsent(entry.getDocumentNumber(), key -> new MutableInt(0));
+        nextSequenceNumber.increment();
+        entry.setTransactionLedgerEntrySequenceNumber(nextSequenceNumber.toInteger());
+      }
+      collectorBatch.setOriginEntries(consolidatedOriginEntries);
+    }
+    LOG.info("consolidateOriginEntries, consolidated number of origin entries: " + consolidatedOriginEntries.size());
+  }
+
+  private OriginEntryFull findMatchingOriginEntryFull(List<OriginEntryFull> entries, OriginEntryFull searchEntry) {
+    for (OriginEntryFull entry : entries) {

```

Comment:
This full-iteration matching could become very inefficient if the SAE file has a large number of rows in it. If there's a way to limit this check to a certain subset (such as groupings by document number) while still preserving the original entry ordering as much as possible, that would help with keeping this code efficient for larger SAE files.

Reply:
If this ends up being infeasible to refactor, or if the Production files are not actually large enough to cause this code to slow down significantly, then we might be okay with moving this forward as-is.

Reply:
I will get some large SAE file and test performance locally.  I'd like to only do some optimization for performance if we really need it.

Reply:
Given the results of your SAE file testing, I'm fine with leaving this consolidation algorithm as-is. Thanks for checking!